test/extensions/formatter/cel/cel_test.cc

```diff
@@ -130,6 +130,22 @@ TEST_F(CELFormatterTest, TestMaxLength) {
   EXPECT_EQ("/original", formatter->formatWithContext(formatter_context_, stream_info_));
 }
 
+TEST_F(CELFormatterTest, TestContains) {
+  const std::string yaml = R"EOF(
+  text_format_source:
+    inline_string: "%CEL(request.url_path.contains('request'))%"
```

Comment:
I'm not sure it's actually possible to parse this well without escaping. Consider an expression `request.path.contains(')')`. A parser needs to be aware of quoting inside CEL, to properly delineate the expression. That's not doable with regex for eBNF grammar. I think you have to require escaping for `)` character?

Reply:
I think both `)` and `%` need escaping.